### DISCUSSION TITLE
how can i use variables as katex functions?

### DISCUSSION THREAD

**Discussion by xed3x** [KaTeX/KaTeX#2094] [Q&A] [answered]
hi. just one question. 
how can i use in 

    katex.render("a + b = c", symbole_40d);

some variables like 

    var a = 2
    var b = 4
    var c = 0
    katex.render("{$a} + {$b} = {$c}", symbole_40d);
    console.log('c=' + c) //Output: c=6

in that kind of meaning. i know it would not works, but for that try to represent that what i need i think its ok as symbolic code :D. big thx

**Comment by edemaine** [comment, marked as answer]
Here are a few ways:
1. using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals):
   ```js
   katex.render(`${a} + ${b} = ${c}`, symbole_40d);
   ```
2. Using `+` operator on strings, like you did in the `console.log` call:
   ```js
   katex.render(a + ' + ' + b + ' = ' + c, symbole_40d);
   ```
3. Using KaTeX macros:
   ```js
   katex.render("\\a + \\b = \\c", symbole_40d, {
     macros: {
       "\\a": a,
       "\\b": b,
       "\\c": c,
     }
   });
   ```
   (Note that this overrides the default meaning of `\c` (a type of accent). It might be better to call the macro something else.)

**Comment by xed3x** [comment]
big thx!
